core/perf_test/Benchmark_ViewCopy.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_CORE_PERFTEST_BENCHMARK_VIEW_COPY_HPP
+#define KOKKOS_CORE_PERFTEST_BENCHMARK_VIEW_COPY_HPP
+
+#include <Kokkos_Core.hpp>
+#include <benchmark/benchmark.h>
+#include <cmath>
+
+namespace {
+
+void report_results(benchmark::State& state, double time) {
+  state.SetIterationTime(time);
+
+  const auto N8        = std::pow(state.range(0), 8);
+  const auto size      = N8 * 8 / 1024 / 1024;
+  state.counters["MB"] = benchmark::Counter(size, benchmark::Counter::kDefaults,
+                                            benchmark::Counter::OneK::kIs1024);
+  state.counters["FOM: GB/s"] =
```

Comment:
Providing the Figure of Merit marker manually - I'll check if this can be done in some better way.

Reply:
This could be as simple as:
```cpp
namespace Test {
std::string FOM(const std::string& label) { return "FOM: " + label; }
}
``` 
So we would have:
```cpp
   state.counters[FOM("GB/s")] = 
```